### PR TITLE
add CI steps to build go assets for mac and linux, add install script, and document install steps

### PR DESCRIPTION
**Pull request by ramfox**
Additional testing will likely be needed to ensure the release process (`release.yml`) and download script (`InstallGo`) work as expected once the PR in merged, but we can now build static libraries that can be provided without additional ENV flags when using golang.

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Python Wheels
 on:
   push:
     branches:
-      - main
+      - main
```
Comment (dignifiedquire) [inline .github/workflows/wheels.yml:6]: I am not needed

File: .gitignore [hunk 2]
```diff
@@ -106,7 +106,7 @@ docs/_build/
 
 # Go
 iroh-node-go
-/iroh-go/iroh/ffi/
+/iroh-go/iroh/libiroh.a
```
Comment (dignifiedquire) [inline .gitignore:109]: I think there is a different file extension on windows?

File: Cargo.toml [hunk 3]
```diff
@@ -14,6 +14,9 @@ documentation = "https://iroh.computer/docs"
 # Sadly this also needs to be updated in .github/workflows/ci.yml
 rust-version = "1.67"
 
+[profile.release]
+strip = "symbols"
```
Comment (dignifiedquire) [inline Cargo.toml:18]: should we really be doing this? cc @Arqu

File: README.go.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# iroh-go
+
+Instructions on how to use `iroh-ffi/iroh-go` in your project!
+
+```
+$ go get github.com/n0-computer/iroh-ffi/iroh-go/iroh
+$ git submodule add https://github.com/n0-computer/iroh-ffi.git extern/iroh-ffi
+$ make -f extern/iroh-ffi/InstallGo
```
Comment (dignifiedquire) [inline README.go.md:8]: does this require a working rust toolchain?

Comment (ramfox) [reply README.go.md:8]: no, this should download the libiroh file from our releases page & slot it into the expected location. No need to have rust.

File: Cargo.toml (hunk 3 above)
Comment (ramfox) [reply Cargo.toml:18]: I added that in a bid to try to reduce the size of the static linked file, but it actually has no effect now that I'm testing it with a clearer head.

Took it out.

File: .gitignore (hunk 2 above)
Comment (ramfox) [reply .gitignore:109]: adjusted it, but also planning to do a mini push for windows after this PR, now that i have access to one again.

File: README.go.md (hunk 4 above)
Comment (dignifiedquire) [reply README.go.md:8]: great

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**